packages/plugin-avatar/src/avatar.js

```diff
@@ -0,0 +1,113 @@
+/**!
+ *
+ * Copyright (c) 2015-2016 Cisco Systems, Inc. See LICENSE file.
+ * @private
+ */
+
+import AvatarUrlBatcher from './avatar-url-batcher';
+import AvatarUrlStore from './avatar-url-store';
+import {SparkPlugin} from '@ciscospark/spark-core';
+import {User} from '@ciscospark/plugin-user';
+import {defaults} from 'lodash';
+
+const Avatar = SparkPlugin.extend({
+  namespace: `Avatar`,
+
+  children: {
+    batcher: AvatarUrlBatcher
+  },
+
+  session: {
+    store: {
+      default() {
+        return new AvatarUrlStore();
+      },
+      type: `any`
+    }
+  },
+
+  /**
+   * @private
+   * Requests an avatar URL from the api
+   * @param {string} uuid
+   * @param {Object} options
+   * @param {integer} options.size
+   * @param {integer} options.cacheControl
+   * @returns {Promise<Object>}
+   */
+  _fetchAvatarUrl(uuid, options) {
+    return this.store.get(uuid, options.size)
+      .catch(() => this.batcher.request(Object.assign({}, {uuid, size: options.size}))
+        .then((item) => this.store.add(defaults(item,
+                                                {cacheControl: options.cacheControl}))
+        )
+      );
+  },
+
+  /**
+   * Retrieves an Avatar from a cache or the api on misses.
+   *
+   * Avatars are square images with sides of 1600, 640, 192, 135, 110, 80, 50,
+   * or 40 pixels. If no size is specified, 80px will be retrieved. If a
+   * non-standard size is requested, the server will return the closest-but-
+   * greater size (or 1600 if request is larger).
+   *
+   * @param {UserObject|string} user The user, Spark user uuid, or email
+   * @param {Object} [options]
+   * @param {integer} [options.size] In {1600, 640, 192, 135, 110, 80, 50, 40}
+   *                                 Defaults to 80 if falsy
+   * @returns {Promise<string>} A promise that resolves to the avatar
+   */
+  retrieveAvatarUrl(user, options) {
+    /* eslint complexity: [0] */
+    options = defaults(options, {size: this.config.defaultAvatarSize,
+                                 cacheControl: this.config.cacheExpiration});
+
+    return User.asUUID(user)
```

Comment:
The user plugin should be on the spark object. I just tested this out after discussion with @ianwremmel and it seems to work properly: `return this.spark.user.asUUID(user)`